File: transfere.js
const prod = [16320, 23400, 26440, 28790, 36540];
const estoque = [344, 1435, 2899, 310, 431];
const quantMin = [200, 500, 800, 150, 100];
const quantVendas = [153, 937, 214, 233, 617];
const estAposVenda = [];
const neces = [];
const transf = [];

console.log("Necessidade de Transferência Armazém para CO");

console.log(
    "\nProduto QtCO    QtMin   QtVendas   Estq.após Vendas  Necess.  Transf. de Arm p/ CO\n"
);
for (let i = 0; i < prod.length; i++) {
    estAposVenda[i] = estoque[i] - quantVendas[i];
    if (estAposVenda[i] < quantMin[i]) {
        neces[i] = quantMin[i] - estAposVenda[i];

        if (neces[i] > 1 && neces[i] < 10) {
            transf[i] = 10;
        } else {
            transf[i] = neces[i];
        }
    } else {
        neces[i] = 0;
        transf[i] = 0;
    }
    console.log(
        `${prod[i]}   ${estoque[i]}    ${quantMin[i]}      ${quantVendas[i]}             ${estAposVenda[i]}              ${neces[i]}           ${transf[i]}`
    );
}

/* SAÍDA DO PROGRAMA 
Necessidade de Transferência Armazém para CO

Produto QtCO    QtMin   QtVendas   Estq.após Vendas  Necess.  Transf. de Arm p/ CO

16320   344    200      153             191              9           10
23400   1435    500      937             498              2           10
26440   2899    800      214             2685              0           0
28790   310    150      233             77              73           73
36540   431    100      617             -186              286           286*/

File: Desafio/SolucaoDesafio/SolucaoDesafio/Entidades/Transfere.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Desafio.Entidades
{
    class Transfere
    {
        public int QtCO { get; set; }
        public int QtMin { get; set; }
        public int QtVendas { get; set; }
        public int EstoqueAposVendas { get; set; }
        public int Necess { get; set; }
        public int TransfArmazemCO { get; set; }

        public Transfere()
        {
        }

        public void ArquivoTransfere(List<Produto> produtos, List<Vendas> vendas)
        {
            const string caminhoArquivoTransfere = @"C:\Projects\quero-ser\Desafio\SolucaoDesafio\SolucaoDesafio\Arquivos\transfere.txt";
            StringBuilder sbCabecalho = new StringBuilder();
            StringBuilder sbDados = new StringBuilder();

            using (StreamWriter writer = new StreamWriter(caminhoArquivoTransfere))
            {

                writer.WriteLine("Necessidade de Transferência Armazém para CO" + Environment.NewLine);

                sbCabecalho.AppendFormat("{0,-9}{1,-8}{2,-8}{3,-15}{4,-25}{5,-10}{6,-25}{7}",
                                         "Produto",
                                         "QtCO",
                                         "QtMin",
                                         "QtVendas",
                                         "Estq. após Vendas",
                                         "Necess.",
                                         "Transf. de Arm p/ CO",
                                         Environment.NewLine);

                writer.WriteLine(sbCabecalho.ToString());

                foreach (Produto produto in produtos)
                {
                    foreach (Vendas venda in vendas)
                    {
                        if (produto.CodProduto == venda.CodProduto && venda.SitVenda == "100" || produto.CodProduto == venda.CodProduto && venda.SitVenda == "102")
                        {
                            QtVendas += int.Parse(venda.QtdVendida);
                        }
                    }
                    
                    QtCO = int.Parse(produto.QtdEstoque);
                    QtMin = int.Parse(produto.QtdMinimaCO);
                    EstoqueAposVendas = QtCO - QtVendas;

                    if (EstoqueAposVendas > QtMin)
                    {
                        Necess = 0;
                        TransfArmazemCO = Necess;
                    }
                    
                    if (EstoqueAposVendas < QtMin)
                        Necess = QtMin - EstoqueAposVendas;

                    if (Necess > 1 && Necess < 10)
                        TransfArmazemCO = 10;
                    else
                        TransfArmazemCO = Necess;

                    sbDados.AppendFormat("{0,-9}{1,-8}{2,-8}{3,-15}{4,-25}{5,-10}{6,-25}{7}",
                                         produto.CodProduto,
                                         QtCO,
                                         QtMin,
                                         QtVendas,
                                         EstoqueAposVendas,
                                         Necess,
                                         TransfArmazemCO,
                                         Environment.NewLine);

                    QtVendas = 0;
                }
                writer.WriteLine(sbDados.ToString());
            }
        }

    }
}
